File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        left = 0
        right = 0
        max_len = 0
        seen_till_now = set()

        while right < len(s):
            while s[right] in seen_till_now:
                seen_till_now.remove(s[left])
                left += 1
            seen_till_now.add(s[right])
            max_len = max(max_len, right - left + 1)
            right += 1
        
        return max_len
        

File: Data Structures & Algorithms/max-water-container/submission-2.java
class Solution {
    public int maxArea(int[] heights) {
        int left = 0, right = heights.length - 1;

        int maxArea = 0;

        while(right > left){
            maxArea = Math.max(maxArea, (right - left) * Math.min(heights[left], heights[right]));

            if(heights[left] > heights[right]){
                right--;
            }else{
                left++;
            }
        }

        return maxArea;
    }
}
